application/src/main/java/org/togetherjava/tjbot/commands/moderation/WarnCommand.java

```diff
@@ -76,45 +97,38 @@ public WarnCommand(@NotNull ModerationActionsStore actionsStore) {
                 target, dmNoticeText, reason);
     }
 
-    private boolean handleChecks(@NotNull Member bot, @NotNull Member author,
-            @NotNull CharSequence reason, @NotNull Guild guild, @NotNull Interaction event) {
-
-        if (!ModerationUtils.handleHasAuthorRole(ACTION_VERB, hasRequiredRole, author, event)) {
-            return false;
-        }
-        if (!ModerationUtils.handleHasBotPermissions(ACTION_VERB, Permission.BAN_MEMBERS, bot,
-                guild, event)) {
-            return false;
-        }
-        if (!ModerationUtils.handleHasAuthorPermissions(ACTION_VERB, Permission.BAN_MEMBERS, author,
-                guild, event)) {
-            return false;
-        }
-        return ModerationUtils.handleReason(reason, event);
-    }
-
     @Override
     public void onSlashCommand(@NotNull SlashCommandEvent event) {
-        OptionMapping userOption =
-                Objects.requireNonNull(event.getOption(USER_OPTION), "The option is null");
-        User target = userOption.getAsUser();
+        OptionMapping targetOption =
+                Objects.requireNonNull(event.getOption(USER_OPTION), "The target is null");
+        User target = targetOption.getAsUser();
```

Comment:
As I said earlier, this is obsolete. You only need the `Member` and thats it.

Reply:
For the ban command you have `OptionMapping targetOption =
                Objects.requireNonNull(event.getOption(TARGET_OPTION), "The target is null");
        User target = targetOption.getAsUser();
 `
 
 I need the member and the user for this command so i need OptionMapping.

Reply:
But you dont need them multiple times and you can get the user out of the member. So you only need the member as variable.

Reply:
The code is now inconsistent. Either you create variables for both `User` and `Member` or you inline both calls using `targetOption`.